examples/radiation_transport/radiation_calculation.py

```diff
@@ -0,0 +1,194 @@
+from bluemira.equilibria import Equilibrium
+from bluemira.geometry._deprecated_loop import Loop
+from bluemira.radiation_transport.advective_transport import ChargedParticleSolver
+from bluemira.radiation_transport.radiation_profile import STCore, STScrapeOffLayer
+
+# %%[markdown]
+# Read equilibrium
+
+# %%
+read_path = get_bluemira_path("equilibria", subfolder="data")
+eq_name = "DN-DEMO_eqref.json"
+eq_name = os.sep.join([read_path, eq_name])
+eq = Equilibrium.from_eqdsk(eq_name, load_large_file=True)
+
+# %%[markdown]
+# Get first wall shape
+
+# %%
+read_path = get_bluemira_path("radiation_transport/test_data", subfolder="tests")
+fw_name = "DN_fw_shape.json"
+fw_name = os.sep.join([read_path, fw_name])
+fw_shape = Loop.from_file(fw_name)
+
+# %%[markdown]
+# Run particle solver
+
+# %%
+p_solver_params = ParameterFrame()
+solver = ChargedParticleSolver(p_solver_params, eq, dx_mp=0.001)
+solver.analyse(first_wall=fw_shape)
+
+# %%[markdown]
+# Run PROCESS solver
+
+# %%
+PROCESS_PATH = ""
+binary = f"{PROCESS_PATH}process"
+
+new_params = {
+    "kappa": 1.6969830041844367,
+}
+params = Configuration(new_params)
+
+build_config = {
+    "mode": "run",
+    "binary": binary,
+}
+
+process_solver = process.Solver(
+    params=params,
+    build_config=build_config,
+)
+process_solver.run()
+
+# %%[markdown]
+# Get impurity fractions
+
+# %%
+impurity_content = {
+    "H": process_solver.get_species_fraction("H"),
+    "He": process_solver.get_species_fraction("He"),
+    "Xe": process_solver.get_species_fraction("Xe"),
+    "W": process_solver.get_species_fraction("W"),
+}
+
+# %%[markdown]
+# Get impurity data: temperature reference
+# and radiative loss function reference
+
+# %%
+impurity_data = {
+    "H": {
+        "T_ref": process_solver.get_species_data("H")[0],
+        "L_ref": process_solver.get_species_data("H")[1],
+    },
+    "He": {
+        "T_ref": process_solver.get_species_data("He")[0],
+        "L_ref": process_solver.get_species_data("He")[1],
+    },
+    "Xe": {
+        "T_ref": process_solver.get_species_data("Xe")[0],
+        "L_ref": process_solver.get_species_data("Xe")[1],
+    },
+    "W": {
+        "T_ref": process_solver.get_species_data("W")[0],
+        "L_ref": process_solver.get_species_data("W")[1],
+    },
+}
+
+# %%[markdown]
+# Eventually customise plasma parameters to deviate from default values
+
+# %%
+plasma_params = ParameterFrame(
+    # fmt: off
+    [
+        ["kappa", "Elongation", 3, "dimensionless", None, "Input"],
+    ]
+    # fmt: on
+)
+
+# %%[markdown]
+# Eventually customise radiation parameters to deviate from default values
+
+# %%
+rad_params = ParameterFrame(
+    # fmt: off
+    [
+        ["p_sol", "power entering the SoL", 300e6, "W", None, "Input"],
+    ]
+    # fmt: on
+)
+
+# %%[markdown]
+# Run core radiation source calculation for Spherical Tokamak
+
+# %%
+stcore = STCore(solver, impurity_content, impurity_data, plasma_params, rad_params)
```

Comment:
The discretisation used in this examples seems to be hard-coded to the grid of the equilibrium and some lin-spacing (hardcoded to 50 points) and looks extremely weird...

Reply:
This discretisation comes from flux_surfaces.py and in turn from equilibrium.py.
I just wanted to be consistent with the discretisation that is used in advective_transport.py.

Reply:
So, for sure having the option of chosing the number of points for the discretisation is a nice feature, but I think it would be useful just for a nicer plot, more continuous.
Here, I guess the goal is to calculate a function for the emission source that would serve as input for a tracing code (e.g. CHERAB). 
Indeed, the missing final step here (this would eventually anwer part of the question in the heading), is to generate a grid from which extrapolating the said function. 
Also, talking about grids, the scrape off layer one, is not an actual grid, or at least not made purposely. It is just the result of the points discretising each flux surface. I didn't set any grid, yet.